### PR TITLE
Feature/5279 5368 event checkin persistence

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR adds the implementation of the EventStore.
It also has some small improvements for the ContactDiaryStore.

Please note: Because there are many changes, Github will collapse the change of the EventStore and EventStoreTests.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5655

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	let id: String
+	let type: EventType
+	let description: String
+	let address: String
+	let start: Date
+	let end: Date
+	let defaultCheckInLengthInMinutes: Int
+	// The signature of the event (provided by the CWA server). It is stored as a base64-encoded string of the signature attribute of Protocol Buffer message SignedEvent.
+	let signature: String
+}
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift:22]: It's event redundant to traceLocation? Maybe this is an old file?

**Review by KaiTeuber — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift]: Good point! Removed old file.

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/ContactDiaryStoreTests.swift [hunk 2]
```diff
@@ -852,6 +852,10 @@ class ContactDiaryStoreTests: XCTestCase {
 		let databaseQueue = makeDatabaseQueue()
 		let store = makeContactDiaryStore(with: databaseQueue)
 
+		databaseQueue.inDatabase { database in
+			XCTAssertEqual(database.numberOfTables, 4, "Looks like there is a new table. Please extend this test and add the new table to the dropTables() function.")
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/ContactDiaryStoreTests.swift:856]: 👍

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift [hunk 3]
```diff
@@ -10,12 +10,12 @@ struct Checkin {
...
-	let traceLocationDefaultCheckInLengthInMinutes: Int
+	let traceLocationStartDate: Date?
+	let traceLocationEndDate: Date?
+	let traceLocationDefaultCheckInLengthInMinutes: Int?
 	let traceLocationSignature: String
 	let checkinStartDate: Date
-	let checkinEndDate: Date
-	let targetCheckinEndDate: Date
+	let checkinEndDate: Date?
+	let targetCheckinEndDate: Date?
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift:19]: If I understand it correctly, this field should not be optional.

At least according to [EXPOSUREAPP-5582](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5582): "Automatic check-out time is mandatory and must be later than current time"

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift:19]: According to the tech spec it is optional. https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/event-registration-mvp/docs/spec/event-registration-client.md#database-table-for-checkins

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift:19]: I ask max about that.

**Review by nickguendling — APPROVED** [review]

**Pull request merged**